File: server/shared/types/menu.go
package types

type MenuPayload struct {
	Name        string   `form:"name" binding:"required"`
	Price       float64  `form:"price" binding:"required"`
	Category    string   `form:"category" binding:"required"`
	Description string   `form:"description"`
	Ingredients []string `form:"ingredients"`
}

type MenuEditPayload struct {
	Name        *string   `form:"name"`
	Category    *string   `form:"category"`
	Price       *float64  `form:"price" validate:"omitempty,gt=0"`
	Status      *bool     `form:"status"`
	Description *string   `form:"description"`
	Ingredients *[]string `form:"ingredients"`
}

File: web/src/types/menu.ts
export interface Menu {
  id: number
  name: string
  description?: string
  price: number
  status: boolean
  picture: string
  category: string
  ingredients?: Array<string>
}

export interface TopSellingItem {
  id: number
  name: string
  revenue: number
  total_sold: number
}

export interface OrdersByCategory {
  period: string
  category: string
  revenue: number
  total_sold: number
}

export interface OrdersAndRevenue {
  period: string
  total_orders: number
  total_revenue: number
}
